### PR TITLE
fix: fix upgrade

### PR DESCRIPTION
**Pull request by niladrih**
Change:
- Use yq's strenv operator to use special characters with yq when updating the upgrade values.yaml.
- Fix check which forbids upgrade from a stable release to unstable version.
- Prefers the parameters of an existing default StorageClass over the chart defaults.

### CODE REVIEW

File: chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -205,3 +205,28 @@ Usage:
...
+    {{- $sc := lookup "storage.k8s.io/v1" "StorageClass" "" $scName -}}
+    {{- if $sc -}}
+        {{ range $param, $val := $sc.parameters }}
+{{ $param | quote }}: {{ $val | quote }}
+        {{- end -}}
+    {{- else -}}
+{{/*
+    Latest defaults
+*/}}
+protocol: {{ $valuesProtocol | quote }}
```
Comment (tiagolobocastro) [inline chart/templates/_helpers.tpl:229]: this we could take from values.yaml and avoid modifying the template again in the future

File: k8s/upgrade/src/plugin/preflight_validations.rs [hunk 2]
```diff
@@ -186,6 +186,16 @@ impl TryFrom<&[u8]> for UnsupportedVersions {
...
 }
 
+/// Strips the prefix 'v' from a semver-like literal, e.g.: v1.2.3 -> 1.2.3.
+/// The Version crate doesn't work with the 'v' prefix.
+pub(crate) fn strip_v_prefix(version: &str) -> &str {
+    if version.starts_with('v') {
+        return version.strip_prefix('v').unwrap_or(version);
+    }
+
+    version
```
[the remark this answers is not in the document]
Comment (niladrih) [reply k8s/upgrade/src/plugin/preflight_validations.rs]: Made this change.

File: chart/templates/_helpers.tpl (hunk 1 above)
Comment (niladrih) [reply chart/templates/_helpers.tpl]: Good one, made this change.

File: chart/templates/_helpers.tpl [hunk 3]
```diff
@@ -205,3 +205,30 @@ Usage:
...
+    {{- if $sc -}}
+        {{/* Existing defaults */}}
+        {{ range $param, $val := $sc.parameters }}
+{{ $param | quote }}: {{ $val | quote }}
+        {{- end -}}
+
+    {{- else -}}
+        {{/* Current defaults */}}
+        {{ range $param, $val := $valuesParams }}
+{{ $param | quote }}: {{ $val | quote }}
```
Comment (tiagolobocastro) [inline chart/templates/_helpers.tpl:231]: Does param need quote?

**Review by tiagolobocastro — APPROVED** [review]

File: chart/templates/_helpers.tpl (hunk 3 above)
Comment (niladrih) [reply chart/templates/_helpers.tpl:231]: Yes, I think so. It is a map[string]string.

**Review by w3aman — APPROVED** [review]
Tested this upgrade on k8s cluster

**Pull request merged**